lib/linux_admin/network_interface.rb

```diff
@@ -0,0 +1,183 @@
+    def netmask
+      @network_conf[:mask]
+    end
+
+    # Retrieve the IPv6 sub-net mask assigned to the interface
+    #
+    # @return [String] IPv6 netmask
+    # @raise [ArgumentError] if the given scope is not `:global` or `:link`
+    def netmask6(scope = :global)
+      if scope == :global
+        @network_conf[:mask6_global]
+      elsif scope == :link
+        @network_conf[:mask6_link]
+      else
+        raise ArgumentError, "Unrecognized address scope #{scope}"
+      end
+    end
+
+    # Retrieve the IPv4 default gateway associated with the interface
+    #
+    # @return [String] IPv4 gateway address
+    def gateway
+      @network_conf[:gateway]
+    end
+
+    # Brings up the network interface
+    #
+    # @return [Boolean] whether the command succeeded or not
+    def ifup
+      run(cmd("ifup"), :params => [@interface]).success?
+    end
+
+    # Brings down the network interface
+    #
+    # @return [Boolean] whether the command succeeded or not
+    def ifdown
+      run(cmd("ifdown"), :params => [@interface]).success?
+    end
+
+    private
+
+    # Parses the output of `ip addr show`
+    #
+    # @param output [String] The command output
+    # @param regex  [Regexp] Regular expression to match the desired output line
+    # @param col    [FixNum] The whitespace delimited column to be returned
+    # @return [String] The parsed data
+    def parse_ip_output(output, regex, col)
+      the_line = output.split("\n").detect { |l| l =~ regex }
+      the_line.nil? ? nil : the_line.strip.split(' ')[col]
+    end
+
+    # Runs the command `ip addr show <interface>`
+    #
+    # @return [String] The command output, nil on failure
+    def ip_show
+      result = run(cmd("ip"), :params => ["addr", "show", @interface])
+      result.success? ? result.output : nil
+    end
+
+    # Parses the IPv4 information from the output of `ip addr show <device>`
+    #
+    # @param ip_output [String] The command output
+    def parse_ip4(ip_output)
+      cidr_ip = parse_ip_output(ip_output, /inet/, 1)
+      return unless cidr_ip
+
+      @network_conf[:address] = cidr_ip.split('/')[0]
+      @network_conf[:mask] = IPAddr.new('255.255.255.255').mask(cidr_ip.split('/')[1]).to_s
+    end
+
+    # Parses the IPv6 information from the output of `ip addr show <device>`
+    #
+    # @param ip_output [String] The command output
+    # @param scope     [Symbol] The IPv6 scope (either `:global` or `:local`)
+    def parse_ip6(ip_output, scope)
+      mask_addr = IPAddr.new('ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff')
+      cidr_ip = parse_ip_output(ip_output, /inet6 .* scope #{scope}/, 1)
+      return unless cidr_ip
+
+      parts = cidr_ip.split('/')
+      @network_conf["address6_#{scope}".to_sym] = parts[0]
+      @network_conf["mask6_#{scope}".to_sym] = mask_addr.mask(parts[1]).to_s
+    end
+  end
+end
+
+Dir.glob(File.join(File.dirname(__FILE__), "network_interface", "*.rb")).each { |f| require f }
```

Comment:
`File.dirname(__FILE__)` can be replaced with `__dir__` (or do we still support old rubies?)